### PR TITLE
DOCS - replacing the default authenticator

### PR DESCRIPTION
**Pull request by antonsmithnz**
Please see #23 for a description of this PR :)

I also noticed that some of the nav links were broken too, I've addressed this in the PR as well.

### CODE REVIEW

File: docs/en/developer.md [hunk 1]
```diff
@@ -117,6 +119,32 @@ if (getenv('LDAP_HOSTNAME') && getenv('LDAP_USERNAME') && getenv('LDAP_PASSWORD'
 }
 ```
 
+### Set LDAP as default authenticator
+
+If you’d like to set LDAP as the default authenticator, you will need to reset the list of authenticators first. You can achieve this with the following approach in the `mysite/_config/ldap.yml` file:
+
+```
+---
+Name: authreset
```
Comment (ScopeyNZ) [inline docs/en/developer.md:128]: This block isn't actually required as the `default` authenticator is being specifically overwritten (eg. https://github.com/silverstripe/silverstripe-mfa/blob/master/_config/config.yml#L1-L10)

**Review by ScopeyNZ — CHANGES_REQUESTED** [review]

**Comment by ScopeyNZ** [comment]
Thanks for the contribution! While what you've suggested would work well, I think we should probably suggest an example without the "reset" hackiness if we can avoid it.

File: docs/en/developer.md (hunk 1 above)
Comment (antonsmithnz) [reply docs/en/developer.md:128]: Normally I would agree with you, but when I tried it without the reset block, the member authenticator was still the default, even after /dev/build?flush=1

**Comment by antonsmithnz** [comment]
No worries, see comments. I can remove the reset authenticator block if you prefer.

**Comment by lozcalver** [comment]
~Can confirm the need to add a “reset” YAML block to remove the built-in authenticator, I had to do the exact same for https://github.com/bigfork/silverstripe-oauth-login#replacing-the-default-authenticator~

Edit: actually, I think I needed to do that in my module because my authenticator has a name other than `default`. @ScopeyNZ is right - naming the key `default` _should_ override it

File: docs/en/developer.md (hunk 1 above)
Comment (antonsmithnz) [reply docs/en/developer.md:128]: Following kinglozzer's comments, I'll remove the reset block :)

**Comment by antonsmithnz** [comment]
I'm a little confused now. The YML snippet I added is exactly the same as the YML one in this section https://github.com/silverstripe/silverstripe-ldap/blob/master/docs/en/developer.md#show-the-ldap-login-button-on-login-form. 

So now I'm wondering if I need to add my PR at all.

I should point out that when I tried this as written originally, I never saw an LDAP button on the login form as described. After /dev/build flush and then logging in, it still used the member authenticator.

I'm no longer working on the project using this module so I won't have time to go back and test some things unfortunately.

I have a question - Should an LDAP button be showing on the login form after adding this yml https://github.com/silverstripe/silverstripe-ldap/blob/master/docs/en/developer.md#show-the-ldap-login-button-on-login-form (is the current doc correct)? 
- If yes, it just seems odd the yml config documented for adding the ldap button is the same for making the ldap authenticator the default. Or is it just me?
- If no, should the header "Show the LDAP Login button on login form" be renamed to "Set LDAP as default authenticator"?

**Comment by ScopeyNZ** [comment]
> Should an LDAP button be showing on the login form after adding this yml /docs/en/developer.md@master#show-the-ldap-login-button-on-login-form  (is the current doc correct)?

Yes. 

> it just seems odd the yml config documented for adding the ldap button is the same for making the ldap authenticator the default. Or is it just me?

It does seem a little odd. The documentation seems to make the assumption that you will be using LDAP as a replacement authenticator - not as an alternative.

I think the docs can be a little clearer here, offering two alternatives for "replace" and "additonal". The additional one should just change `default:` to something else like `ldap:`.

**Comment by robbieaverill** [comment]
Perhaps your issue with it not overloading the `default` authenticator key was caused by config block priority (e.g. `Before: '#someconfig'` and `After: '#anotherconfig'`)?

Side note: were those TOC updates part of another change? We could keep those if so =)

**Comment by emteknetnz** [comment]
This pull request hasn't had any activity for a while. Are you going to be doing further work on it, or would you prefer to close it now?

**Comment by antonsmithnz** [comment]
Let's close it. I haven't done SS dev in a while and won't be revisiting this anytime soon :)

**Pull request closed without merging**